src/components/PrivateRoute/PrivateRoute.jsx

```diff
@@ -0,0 +1,20 @@
+import { Redirect, Route, useLocation } from "react-router-dom";
+import { useUser } from "../../context/UserContext";
+
+export default function PrivateRoute({ children, ...rest }) {
+  const { user } = useUser();
+  const location = useLocation();
+
+  return (
+    <Route {...rest}>
+      {user.email ? (children)
+      : (
+        <Redirect to={{
+          pathname: './login',
+          state: { origin: location },
```

Comment:
We actually don't need the period (.) in pathname='./login' (has to match App.js path within a route).